client/apps.go

```diff
@@ -0,0 +1,88 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015-2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+)
+
+type AppInfo struct {
+	Snap         string `json:"snap,omitempty"`
+	Name         string `json:"name"`
+	DesktopFile  string `json:"desktop-file,omitempty"`
+	*ServiceInfo `json:",omitempty"`
+}
+
+// IsService returns true if the application is a background daemon.
+func (a *AppInfo) IsService() bool {
+	if a == nil {
+		return false
+	}
+	if a.ServiceInfo == nil {
+		return false
+	}
+	if a.ServiceInfo.Daemon == "" {
+		return false
+	}
+
+	return true
+}
+
+type ServiceInfo struct {
+	Daemon          string `json:"daemon"`
+	ServiceFileName string `json:"service-file-name"`
+	Enabled         bool   `json:"enabled"`
+	Active          bool   `json:"active"`
+}
+
+// AppInfoWanted can be used to specify what kind of app is of
+// interest.
+type AppInfoWanted struct {
+	Services bool
+	Commands bool
```

Comment:
Similarly to the prior point, we didn't really cover those distinctions much, and I'm concerned about the line we're drawing implicitly here.  It feels like these changes are starting to create some conflicting terminology on what today is a pretty sane model. For example, every application actually specifies a command. Those things which hold the "daemon" field do not expose said command, but that sounds like a silly limitation which we should drop. So, what would it mean to say "commands" when everything is an application, and everything has commands, and everything should get a command too?

Luckily, we probably don't even need to get to deep into this discussion right away. We might simply change the API to either return everything, or services when AppOption.Service is true. Down the line we can easily add more fields for other filtering options.